ethereum/solidity/docs/contract-explanation.md

```diff
@@ -1,66 +0,0 @@
-# Functioning of the Peggy.sol contract
-
-The Peggy contract locks assets on Ethereum to facilitate a Tendermint blockchain creating synthetic versions of those assets. It is designed to be used alongside software on the Tendermint blockchain, but this article focuses on the Ethereum side.
-
-Usage example:
-
-- You send 25 DAI to the Peggy contract, specifying which address on the Tendermint chain should recieve the syntehtic DAI.
-- Validators on the Tendermint chain see that this has happened and mint 25 synthetic DAI for the address you specified on the Tendermint chain.
-- You send the 25 synthetic DAI to Jim on the Tendermint chain.
-- Jim sends the synthetic DAI to Peggy module on the Tendermint chain, specifying which Ethereum address should receive it.
-- The Tendermint validators burn the synthetic DAI on the Tendermint chain and unlock 25 DAI for Jim on Ethereum
-
-## Security model
-
-The Peggy contract is basically a multisig with a few tweaks. Even though it is designed to be used with a consensus process on Tendermint, the Peggy contract itself encodes nothing about this consensus process. There are three main operations- updateValset, submitBatch, and sendToCosmos. 
-- updateValset updates the signers on the multisig, and their relative powers. This mirrors the validator set on the Tendermint chain, so that all the Tendermint validators are signers, in proportion to their staking power on the Tendermint chain. An updateValset transaction must be signed by 2/3's of the current valset to be accepted.
-- submitBatch is used to submit a batch of transactions unlocking and transferring tokens to Ethereum addresses. It is used to send tokens from Cosmos to Ethereum. The batch must be signed by 2/3's of the current valset.
-- sendToCosmos is used to send tokens onto the Tendermint chain. It simply locks the tokens in the contract and emits an event which is picked up by the Tendermint validators.
-
-### updateValset
```

Comment:
Doesn't make sense to keep stuff related to valsets as we're having the same functionality.
And, maybe in the future, add more docs to cover the tuple roots stuff ?

Reply:
Upon reading that section of the docs, it's basically redundant to the extensive comments that already exist in the contract: https://github.com/celestiaorg/quantum-gravity-bridge/blob/9996dc98551018054306c6344437aa8ce34661a8/ethereum/solidity/src/QuantumGravityBridge.sol#L23-L32

My rule of thumb is: documentation should never be duplicated since it's impossible to properly maintain duplicate docs.

Reply:
agree with John here, but still a good point @SweeXordious and thanks for bringing it up